File: wee2task/radixsort.py
def radix_sort(arr):
    max_num = max(arr)
    exp = 1
    while max_num // exp > 0:
        count = [0] * 10
        for i in range(len(arr)):
            index = arr[i] // exp
            count[index % 10] += 1
        for i in range(1, 10):
            count[i] += count[i-1]
        output = [0] * len(arr)
        i = len(arr) - 1
        while i >= 0:
            index = arr[i] // exp
            output[count[index % 10] - 1] = arr[i]
            count[index % 10] -= 1
            i -= 1
        for i in range(len(arr)):
            arr[i] = output[i]
        exp *= 10
    return arr

File: wee2task/radixsort.java
public class RadixSort {
    public static int[] radixSort(int[] arr) {
        int maxNum = arr[0];
        for (int i = 1; i < arr.length; i++) {
            if (arr[i] > maxNum) {
                maxNum = arr[i];
            }
        }

        int exp = 1;
        while (maxNum / exp > 0) {
            int[] count = new int[10];
            for (int i = 0; i < arr.length; i++) {
                int index = arr[i] / exp;
                count[index % 10]++;
            }
            for (int i = 1; i < 10; i++) {
                count[i] += count[i-1];
            }
            int[] output = new int[arr.length];
            for (int i = arr.length - 1; i >= 0; i--) {
                int index = arr[i] / exp;
                output[count[index % 10] - 1] = arr[i];
                count[index % 10]--;
            }
            for (int i = 0; i < arr.length; i++) {
                arr[i] = output[i];
            }
            exp *= 10;
        }
        return arr;
    }
}
